src/main/java/entities/ExecutionState.java

```diff
@@ -0,0 +1,102 @@
+package entities;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * A high-level entity responsible for representing the execution state of the program. It stores user courses,
+ * next course being generate, user specified base schedule. This class is intended to improve communication between
+ * boundaries in the program.
+ */
+public class ExecutionState {
+    /**
+     * @variable allUserCourses represents all Courses initially instantiated by the user
+     * @variable remainingUserCourses represents all courses the user
+     */
+    private static boolean stateIsSetup;
+    private static GenerationMode generationMode;
+    private static List<Course> allUserCourses;
+    private static List<Course> remainingUserCourses;
+    private static Course currentCourse;
+
+    public ExecutionState(){}
```

Comment:
imo just delete this constructor. It'll be auto-added on compile, and having a constructor (even an empty one) makes me think this class is supposed to be instantiated

Reply:
Yeah true